<?xml version="1.0"?>
<launch>

	<param name="moreo/visualize_process" value="True"/>
	<param name="moreo/visualize_reprojection" value="False"/>


	<!-- tfs -->

	<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera" args="0 0 0.4 0 0 0 base_link camera_link"/> -->
	<!--<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_image_coordinate_system" args="0 0 0 0.5 -0.5 0.5 0.5 worldcoordinates cameracoordinates"/>-->

	 <node type="rviz" name="rviz" pkg="rviz" args="-d $(find moreo)/moreo_rviz.rviz" />
	<!-- If Image input is compressed -->
	<node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/rgb/image_rect_color raw out:=/image_raw" />

	<!-- Nodes to be launched -->

	<node name="Moreo" pkg="moreo" type="moreo_system.py" output="screen"/>

					<!-- Distortion of camera coeff-->
	<param name="cam_dist/dist[0]"    						value="0"/>
	<param name="cam_dist/dist[1]"    						value="0"/>
	<param name="cam_dist/dist[2]"    						value="0"/>
	<param name="cam_dist/dist[3]"    						value="0"/>
	<param name="cam_dist/dist[4]"    						value=" 0"/>

					<!-- Camera_parameters-->
	<param name="image_width"    							value="1440"/>
	<param name="image_height"    						value="1080"/>
	<param name="f_in_pixels"    							value="720"/>
	<param name="hfov"    								value="90"/>
	<param name="cx"		    						value="720"/>
	<param name="cy"    								value="540"/>

						<!-- Published topics -->
	<param name="/moreo/predicted_landmarks" 		value="/Moreo/MoreoLandmarks"/>
	<param name="/moreo/visualize_poses" 			value="Moreo/predicted_poses"/>
	<param name="/moreo/epilines_topic" 			value="moreo/epilines"/>
	<param name="/moreo/matches_topic" 			value="moreo/correct_matches"/>
	<param name="/moreo/reprojection_topic" 		value="moreo/visual_reprojection"/>
	<param name="/moreo/update_bbox" 			value="moreo/updatedBoundingBoxes"/>
						<!-- Subscribed topics -->
	<param name="/moreo/image_raw" 			value="/front_camera_rgb/image_raw"/>
	<param name="/moreo/odom" 				value="/odometry"/>
	<param name="/moreo/bboxes" 				value="/darknet_ros/bounding_boxes_filtered"/>
	<param name="/moreo/gt_relative_cones" 		value="/ObjectList"/>

						<!-- Algorithm parameters -->
	<param name="/moreo/buffer_size" 			value="10"/>
	<param name="/moreo/gap_between_inputs" 		value="8"/>
	<param name="/moreo/gap_between_pairs" 		value="1"/>

</launch>
